2015/proposals/girdhar-yashasvi.md

```diff
@@ -0,0 +1,183 @@
+Until now, we do not collect these results anywhere. But there are number of reasons why we should be doing that.
+
+First of all, it will help us figure out some important information, such as how much trouble the students are having installing some software or which packages need more attention.
+
+Some sample queries that the system would serve will look like:
+
+* how many tests failed in a particular workshop, on windows.
+* what fraction of failed tests were on OS x, for a specific time period.
+* for a particular workshop, find number of failed version tests for git.
+
+It will also help us to make decisions about using the right versions of the dependencies such as deprecating the older versions or adapting with the new versions.
+
+##Technical Details:
+
+The project is mainly divided into four phases:
+
+* Creating a model for storing the data in a relational database.
+
+The first step of the project is deciding what all information is useful to us and how we would be storing it. I have already built a first draft of the schema here, with feedback from the mentor.
+
+* Designing an API to collect the results of the tests.
+
+This step involves designing an API on the server, so that the clients can submit the results through the installation test scripts. The API would collect the results and store them in a 
+relational database, following the schema decided in the first phase.
+I am planning to use Flask + SqlAlchemy for this purpose.
+
+* Modifying the installation test scripts for sending the data to the server.
+
+This step involves sending the required data to the server, in a form that conforms with the API. I have already understood the script and have also submit a PR, that adds documentation explaining the implementation details. 
+
+* API for performing queries on the data (includes building an interface for the target users. )
+
+This step mainly involves three parts : 
+* designing an API to perform queries on the stored data
+* building an interface for the user to use that API, and 
+* adding a data visualisation library for better viewing of the results.
+
+I plan on writing a custom UI in Angularjs that allows the user to construct the queries and then use a data visualisation library for viewing the results.
+
+##Schedule of Deliverables
+
+I plan to complete the back-end before mid term. So that, the latter half will be used for the front end and testing.
+
+##### 27th April to 23rd May : Community bonding period
+
+I would use this period, partially for learning more about the technologies that I would be working on and partially for understanding more about how software carpentry works. I believe that it’s important, so that I believe in its vision and have an inspiration to stay connected with the organisation even after the summer of code ends. I do not want that my work should be a matter of just 3 months.
```

Comment:
The “…period, partially…” comma is a bit awkward.  Maybe “I would use this period for … and …”?  And the usual captialization is “Software Carpentry”.  The doubled “believe” could also be trimmed down to tighten things up.  And maybe “I want my work to last more than three months” or “I want my involvement to last more than three months”, depending on whether you're more focused on writing a maintainable aggregator or on sticking with the larger SWC project in the long term ;).